From: Jiri Slaby <jslaby@suse.cz>
Subject: timer-r0drv-linux: use timer_container_of

from_timer() was renamed to timer_container_of() in 6.16's:
commit 41cb08555c4164996d67c78b3bf1c658075b75f1
Author: Ingo Molnar <mingo@kernel.org>
Date:   Fri May 9 07:51:14 2025 +0200

    treewide, timers: Rename from_timer() to timer_container_of()

---
 src/VBox/Runtime/r0drv/linux/timer-r0drv-linux.c |    4 ++++
 1 file changed, 4 insertions(+)

--- a/src/VBox/Runtime/r0drv/linux/timer-r0drv-linux.c
+++ b/src/VBox/Runtime/r0drv/linux/timer-r0drv-linux.c
@@ -748,7 +748,11 @@ static enum hrtimer_restart rtTimerLinux
  */
 static void rtTimerLinuxStdCallback(struct timer_list *pLnxTimer)
 {
+#if RTLNX_VER_MIN(6,16,0)
+    PRTTIMERLNXSUBTIMER pSubTimer = timer_container_of(pSubTimer, pLnxTimer, u.Std.LnxTimer);
+#else
     PRTTIMERLNXSUBTIMER pSubTimer = from_timer(pSubTimer, pLnxTimer, u.Std.LnxTimer);
+#endif
 #else
 /**
  * Timer callback function for standard timers.
